Library/Homebrew/PATH.rb

```diff
@@ -0,0 +1,73 @@
+class PATH
+  include Enumerable
+  extend Forwardable
+
+  def_delegator :@paths, :each
+
+  def initialize(*paths)
+    @paths = parse(*paths)
+  end
+
+  def prepend(*paths)
+    @paths = parse(*paths, *@paths)
+    self
+  end
+
+  def append(*paths)
+    @paths = parse(*@paths, *paths)
+    self
+  end
+
+  def insert(idx, *paths)
+    @paths = parse(*@paths.insert(idx, *paths))
+    self
+  end
+
+  def select(&block)
+    self.class.new(@paths.select(&block))
+  end
+
+  def reject(&block)
+    self.class.new(@paths.reject(&block))
+  end
+
+  def to_ary
+    @paths
+  end
+  alias to_a to_ary
+
+  def to_str
+    @paths.join(File::PATH_SEPARATOR)
+  end
+  alias to_s to_str
+
+  def ==(other)
+    if other.respond_to?(:to_ary)
+      return true if to_ary == other.to_ary
+    end
+
+    if other.respond_to?(:to_str)
+      return true if to_str == other.to_str
+    end
+
+    false
+  end
+
+  def empty?
+    @paths.empty?
+  end
+
+  def existing
+    existing_path = select(&File.method(:directory?))
+    existing_path unless existing_path.empty?
```

Comment:
The usage for this is more like this:

```ruby
ENV["MY_PATH"] = mypath.existing
```

So, by returning `nil`, this will unset `MY_PATH`, instead of setting it to an empty string.